test/types/decorator.py

```diff
@@ -0,0 +1,31 @@
+from dataclasses import dataclass
+
+import tensorflow as tf
+from beartype import beartype
+
+from jaxtyping import Float, Int, jaxtyped
+
+
+@jaxtyped(typechecker=beartype)
+@dataclass
+class User:
+    name: str
+    age: int
+    items: Float[tf.Tensor, "N"]  # noqa: F821
+    timestamps: Int[tf.Tensor, "N"]  # noqa: F821
```

Comment:
You can skip the `noqa` by adding a space at the start of the shape string, see [this FAQ entry](https://docs.kidger.site/jaxtyping/faq/#flake8-or-ruff-are-throwing-an-error).